nightfall-client/src/services/commitment-sync.mjs

```diff
@@ -0,0 +1,56 @@
+/**
+commitmentsync services to decrypt commitments from transaction blockproposed events 
+or use clientCommitmentSync to decrypt when new ivk is received.
+*/
+
+import config from 'config';
+import logger from 'common-files/utils/logger.mjs';
+import Secrets from '../classes/secrets.mjs';
+import { getAllTransactions } from './database.mjs';
+import { countCommitments, storeCommitment } from './commitment-storage.mjs';
+
+const { ZERO } = config;
+
+/**
+decrypt commitments for a transaction given ivks and nsks.
+*/
+export async function decryptCommitment(transaction, ivk, nsk) {
+  const nonZeroCommitments = transaction.commitments.flat().filter(n => n !== ZERO);
+  ivk.forEach((key, j) => {
+    // decompress the secrets first and then we will decryp t the secrets from this
+    const decompressedSecrets = Secrets.decompressSecrets(transaction.compressedSecrets);
+    logger.info(`decompressedSecrets: ${decompressedSecrets}`);
+    try {
+      const commitment = Secrets.decryptSecrets(decompressedSecrets, key, nonZeroCommitments[0]);
+      if (Object.keys(commitment).length === 0)
+        logger.info("This encrypted message isn't for this recipient");
+      else {
+        // console.log('PUSHED', commitment, 'nsks', nsks[i]);
+        logger.info('This is encrpted and stored');
+        storeCommitment(commitment, nsk[j]);
```

Comment:
I'm not entirely sure if calling `storeCommitment` (which is async) in a `forEach` will work the way we want it. You might need to push into an array that is then return a `Promise.all(...)` at the end or use a `map` and wrap that in a `Promise.all(..)`

Reply:
yeah . It did work in my limited testing . But I have added the necessary changes and pushed new commit.